test/itf/ItfLiquidatorShouldClosePositionTest.ts

```diff
@@ -0,0 +1,963 @@
+            BigNumber.from("6").mul(N0__01_18DEC),
+            BigNumber.from("4").mul(N0__01_18DEC),
+            ZERO,
+            ZERO
+        );
+    });
+
+    it("Should deploy contract", async () => {
+        const testData = await prepareComplexTestDataDaiCase000(
+            BigNumber.from(Math.floor(Date.now() / 1000)),
+            [admin, userOne, userTwo, userThree, liquidityProvider],
+            miltonSpreadModel,
+            PERCENTAGE_5_18DEC
+        );
+
+        const {miltonDai, miltonStorageDai} = testData;
+        const ItfLiquidator = await hre.ethers.getContractFactory("ItfLiquidator")
+        const itfLiquidator = await ItfLiquidator.deploy(miltonDai?.address, miltonStorageDai?.address)
+        // assert deployed
+        expect(await itfLiquidator.deployed()).to.be.not.null;
+    });
+        
+
+    it("should emit CloseSwap event - pay fixed", async () => {
+        //given
+        const testData = await prepareComplexTestDataDaiCase000(
+            BigNumber.from(Math.floor(Date.now() / 1000)),
+            [admin, userOne, userTwo, userThree, liquidityProvider],
+            miltonSpreadModel,
+            PERCENTAGE_5_18DEC
+        );
+
+        const { tokenDai, josephDai, miltonDai, iporOracle, miltonStorageDai } = testData;
+
+
+        if (tokenDai === undefined || josephDai === undefined || miltonDai === undefined) {
+            expect(true).to.be.false;
+            return;
+        }
+
+        const ItfLiquidator = await hre.ethers.getContractFactory("ItfLiquidator")
+        const itfLiquidator = await ItfLiquidator.deploy(miltonDai?.address, miltonStorageDai?.address)
+        const params = getPayFixedDerivativeParamsDAICase1(userTwo, tokenDai);
+
+        await iporOracle
+            .connect(userOne)
+            .itfUpdateIndex(params.asset, PERCENTAGE_5_18DEC, params.openTimestamp);
+
+        await josephDai
+            .connect(liquidityProvider)
+            .itfProvideLiquidity(USD_50_000_18DEC, params.openTimestamp);
+
+        await miltonDai
+            .connect(userTwo)
+            .itfOpenSwapPayFixed(
+                params.openTimestamp,
+                params.totalAmount,
+                params.acceptableFixedInterestRate,
+                params.leverage
+            );
+
+        await miltonDai
+            .connect(userTwo)
+            .itfOpenSwapPayFixed(
+                params.openTimestamp,
+                params.totalAmount,
+                params.acceptableFixedInterestRate,
+                params.leverage
+            );
+
+        await iporOracle
+            .connect(userOne)
+            .itfUpdateIndex(params.asset, PERCENTAGE_160_18DEC, params.openTimestamp);
+
+        // when
+        await expect(
+            itfLiquidator.itfLiquidate([1, 2], [], params.openTimestamp.add(PERIOD_28_DAYS_IN_SECONDS))
+        )
+        //then
+        .to.emit(miltonDai, "CloseSwap")
+        .to.emit(miltonDai, "CloseSwap")
```

Comment:
this is wrong 
if you modify test 

```         // when
        await expect(
            itfLiquidator.itfLiquidate([1], [], params.openTimestamp.add(PERIOD_28_DAYS_IN_SECONDS))
        )
        //then
        .to.emit(miltonDai, "CloseSwap")
        .to.emit(miltonDai, "CloseSwap")
 ```

it doesn't fail,  but it should

You should use something like this 
```
        .to.emit(miltonDai, "CloseSwap").withArgs(1,anyValue,anyValue,anyValue,anyValue,anyValue,anyValue)

```

https://hardhat.org/hardhat-chai-matchers/docs/overview